delfin/drivers/hpe/hpe_msa/ssh_handler.py

```diff
@@ -0,0 +1,465 @@
+import time
+
+import six
+from oslo_log import log as logging
+
+from delfin import exception
+from delfin.common import constants, alert_util
+from delfin.drivers.utils.ssh_client import SSHPool
+from delfin.drivers.utils.tools import Tools
+
+try:
+    import xml.etree.cElementTree as ET
+except ImportError:
+    import xml.etree.ElementTree as ET
+
+
+LOG = logging.getLogger(__name__)
+
+
+class SSHHandler():
+
+    OID_ERR_ID = '1.3.6.1.3.94.1.11.1.1'
+    OID_EVENT_TYPE = '1.3.6.1.3.94.1.11.1.7'
+    OID_LAST_TIME = '1.3.6.1.3.94.1.11.1.4'
+    OID_EVENT_DESC = '1.3.6.1.3.94.1.11.1.9'
+    OID_EVENT_ID = '1.3.6.1.3.94.1.11.1.3'
+    OID_SEVERITY = '1.3.6.1.3.94.1.11.1.6'
+
+    TRAP_SEVERITY_MAP = {
+        '1': 'unknown',
+        '2': 'emergency',
+        '3': 'alert',
+        '4': constants.Severity.CRITICAL,
+        '5': 'error',
+        '6': constants.Severity.WARNING,
+        '7': 'notify',
+        '8': constants.Severity.INFORMATIONAL,
+        '9': 'debug',
+        '10': 'mark'
+    }
+
+    SEVERITY_MAP = {"warning": "Warning",
+                    "informational": "Informational",
+                    "error": "Major"
+                    }
+
+    SECONDS_TO_MS = 1000
+
+    DISK_PHYSICAL_TYPE = {
+        'fc': constants.DiskPhysicalType.FC,
+        'SAS': constants.DiskPhysicalType.SAS
+    }
+
+    def __init__(self, **kwargs):
+        self.ssh_pool = SSHPool(**kwargs)
+
+    def login(self):
+        try:
+            pool_info = self.ssh_pool.do_exec('show pools')
+            if 'is not a recognized command' in pool_info:
+                raise exception.InvalidIpOrPort()
+        except Exception as e:
+            LOG.error("Failed to login msa  %s" %
+                      (six.text_type(e)))
+            raise e
+
+    def get_storage(self, storage_id):
+        try:
+            system_info = self.ssh_pool.do_exec('show system')
+            system_data = self.handle_detail(system_info, 'system')
+            version_info = self.ssh_pool.do_exec('show version')
+            version_arr = self.handle_detail(version_info, 'versions')
+            if version_arr:
+                version_id = version_arr[0]\
+                    .get('bundle-version')
+            if system_data:
+                pool_arr = self.list_storage_pools(storage_id)
+                total_capacity = 0
+                for pool in pool_arr:
+                    total_capacity += int(pool.get('total_capacity'))
+                disk_arr = self.list_storage_disks(storage_id)
+                disk_all = 0
```

Comment:
Why named 'disk_all' instead of 'raw_capacity'?

Reply:
use raw_capacity instead of disk_all